app/lib/actions/middleware/record_smart_proxy_sync_history.rb

```diff
@@ -1,12 +1,32 @@
 module Actions
   module Middleware
     class RecordSmartProxySyncHistory < Dynflow::Middleware
-      def run(*args)
+      def save_smart_proxy_sync_history
         if (action.input[:repository_id] && (action.input[:smart_proxy_id] || action.input[:capsule_id]) && !self.action.output[:smart_proxy_history_id])
-          repo = ::Katello::Repository.find(action.input[:repository_id])
+          repo_id = action.input[:repository_id]
+          repo = ::Katello::Repository.find_by(id: repo_id)
           smart_proxy_id = action.input[:smart_proxy_id] || action.input[:capsule_id]
-          smart_proxy = ::SmartProxy.find(smart_proxy_id)
-          self.action.output[:smart_proxy_history_id] = repo.create_smart_proxy_sync_history(smart_proxy)
+          smart_proxy = ::SmartProxy.find_by(id: smart_proxy_id)
+
+          #skip pulp2 puppet syncs
+          if (repo_pulp_id = action.input[:repo_pulp_id])
+            return if Katello::ContentViewPuppetEnvironment.find_by(pulp_id: repo_pulp_id)
```

Comment:
It seems to be failing without namespacing..
```suggestion
            return if ::Katello::ContentViewPuppetEnvironment.find_by(pulp_id: repo_pulp_id)
```

Reply:
yeah, load order can cause issues on this randomly, fixed!